File: spec/javascripts/models/payment_spec.js
describe('receivable', function(){
	
	describe('A valid receivable object', function(){
		
		it('should be valid given properly formatted values', function(){
			var attrs = factories.payment;
			var payment = new backbone_data.Models.Payment(attrs);
			payment.isValid();
			expect(Object.size(payment.validationError)).toEqual(0);
		});
	
	});
	
	describe('payment validations', function(){
		
		var attrs = factories.payment;
		var payment;
		
		beforeEach(function(){
			payment = new backbone_data.Models.Payment(attrs);	
		});
		
		it('should not be valid if the tenant_id attribute is blank', function(){
			payment.set({tenant_id: ''});
			payment.isValid();
			expect(payment.validationError['tenant_id']).toBeDefined();
		});
		
		it('should not be valid if the receivable_id attribute is blank', function(){
			payment.set({receivable_id: ''});
			payment.isValid();
			expect(payment.validationError['receivable_id']).toBeDefined();
		});
		
		it('should not be valid if the amount attribute is blank', function(){
			payment.set({amount: ''});
			payment.isValid();
			expect(payment.validationError['amount']).toBeDefined();
		});
	
	});
	
});	

File: spec/models/payment_spec.rb
require 'rails_helper'

describe Payment do
  
  describe("-- it should create a payment") do
    
    it("if valid attribute values are given") do
      payment = FactoryGirl.build(:payment)
      expect(payment).to be_valid
    end
  
  end
  
  describe("-- it should not create a payment") do
  
    it("if there is no tenant_id") do
      payment = FactoryGirl.build(:payment, tenant_id: nil)
      payment.valid?
      expect(payment.errors[:tenant_id]).to_not be_blank
    end
  
    it("if there is no receivable_id") do
      payment = FactoryGirl.build(:payment, receivable_id: nil)
      payment.valid?
      expect(payment.errors[:receivable_id]).to_not be_blank
    end
    
    it("if there is no amount") do
      payment = FactoryGirl.build(:payment, amount: nil)
      payment.valid?
      expect(payment.errors[:amount]).to_not be_blank
    end
  
  end

end
